configuration/vpn.md

```diff
@@ -56,3 +59,104 @@ Using a ProxyVM to set up a VPN client gives you the ability to:
 
 5. Optionally, you can install some [custom icons](https://github.com/Zrubi/qubes-artwork-proxy-vpn) for your VPN
+
+**Using iptables and openvpn**
+
+You need an openvpn server and a DNS server accessible through the vpn (use one from your vpn provider / a public one).
+
+1. Create a new VM and check the ProxyVM radio button.
+
+    ![Create\_New\_VM.png](/attachment/wiki/VPN/Create_New_VM.png)
+
+2. Setup openvpn:   
+    Copy your openvpn config file to `/home/user/vpn.cfg`.
+
+    It should have one line starting with `dev` and one starting with `proto`.
+    The first describes the connection type (`tun` or `tap`) and the second the used protocol (`tcp` or `udp`).
+    Depending on your connection type, openvpn will create a new network device (probably `tap0` or `tun0`).
+
+    It also contains a line `remote X.X.X.X 1194`, where `X.X.X.X` is the ip of your openvpn server.
+
+    If it does not contain a line `redirect-gateway def1`, add it.  
+    This will route all traffic through your vpn's network device, after a connection was created.
+    If the connection breaks down all traffic will be routed through the original network device (we will top this with iptables).
+
+    If your vpn config file contains `auth-user-pass`, change it to `auth-user-pass /home/user/auth.txt` and create a file `/home/user/auth.txt` containing the user name in the first line and the password in the second.  
+    This will enable the vpn to login without requiring you to enter your username and password.
+    If a different authentication method is used, set it up to require no user input.  
+    The vpn should now start by calling `sudo openvpn --config /home/user/vpn.cfg` and require no additional user input.  
+
+    In the following, we use the following placeholder:  
+    `$DEV`  For the device created for the connection.  
+    `$PROT` For the protocol used for connection  
+    `$SVR`  For the openvpn server's ip.  
+    `$DNS`  For the dns server's ip.  
+
+
+3.  Setup iptables:  
+    Edit `/rw/config/qubes-firewall-user-script` and add:
+
+    `iptables -P OUTPUT DROP`  
+    This blocks all outgoing traffic, if not specified otherwise.
+    
+    `iptables -I OUTPUT -o $DEV -j ACCEPT`  
+    This allows the local system to connect through the vpn (you dont need this).
+    
+    `iptables -I OUTPUT -o eth0 -d $SVR -p $PROT --dport 1194 -j ACCEPT`  
+    This allows your system to connect to the vpn server with the protocol `$PROT` under the port 1194.
+    
+    `iptables -I OUTPUT -o lo -j ACCEPT`  
+    This allows connections from the system to the system.
+
+    `iptables -I FORWARD -o eth0 -j DROP`  
+    `iptables -I FORWARD -i eth0 -j DROP`  
+    This blocks forwarding of connections through your plain network device (in case the vpn tunnel breaks).
+    
+    `iptables -I FORWARD -o $DEV -j ACCEPT`
+    This allows forwarding of connections through the vpn. (So other AppVMs can use it)
```

Comment:
i guess you mean line 115.
i did not notice, since no appvm behind my vpn currently have rules.
how would you suggest to fix this? (if i omit the line my appvms can't get a connection)